File: routes/PUT/updateProfile.py
from classes import profiles
from flask import make_response, jsonify

def update(profileId, profileData):
    try:
        if not profileData["admin"]:
            return {"message":"You don't have admin privilegies"}

        profile =profiles.Profile(id=profileId,name=profileData["name"],photo=profileData["photo"])
        updateProfile = profile.updateProfile() 
        match updateProfile["message"]:
            case "This profile doesn't exist":
                resp = make_response(updateProfile)
                resp.status_code = 404
                return resp
            case "The profile has been updated":
                resp = make_response(updateProfile)
                resp.status_code = 200
                return resp
            case "An error has occurred while updating the profile":
                resp = make_response(updateProfile)
                resp.status_code = 500
                return resp
    except Exception as e:
        if e.args[0]=="name" or e.args[0]=="photo":
            resp = make_response({"message":"Please, insert the value {}".format(e.args[0])})
            resp.status_code = 400
            return resp
        resp = make_response({"message":"There was an error","error":e.args})
        resp.status_code = 500
        return resp

File: src/routes/PUT/updateProfile.ts
import Profile from "../../classes/profiles";
import type {Response, Request} from 'express'

export default async function updateProfile(req:Request, res:Response){

    try {
        const profileId = Number(req.params.profileId)
        const {name,photo} = req.body
        const update = await new Profile(profileId,name,photo,0).update()
        switch(update.message){
            case "success":
                res.statusCode = 200
                res.json(update)
                break
            case "An error has occurred while updating the profile":
                res.statusCode = 500
                res.json(update)
        }
    } catch (error:any) {
        res.statusCode = 500
        res.json({message:"there was an error updating profile", error:error.message})
    }
}